### ISSUE TITLE
Never exits gracefully

### ISSUE THREAD

**Issue by kirbyclover** [nats-io/java-nats-streaming#26]
Hello again,

Thank you for pointing me to the NATS streaming client.

I'm using version 0.1.2 and following the examples here on the readme.

I can't seem to ever get the client to exit gracefully. I assume some resource or thread pool is still open, but I can't figure out how. 

Here is a simple example. When you run this, the process just hangs

```
import io.nats.stan.Connection;
import io.nats.stan.ConnectionFactory;

public class NatsConnectTest {

  public static void main(String... args) throws Exception {
    final ConnectionFactory connectionFactory = new ConnectionFactory("test-cluster", NatsConnectTest.class.getSimpleName());
    final Connection conn = connectionFactory.createConnection();
    System.out.print("connected");
    conn.close();
    System.out.print("disconnected");
  }
}

```

**Comment by jeffkirby** [comment]
I see the problem. The heart beat subscription is still open in `ConnectionImpl`.

I fixed my local copy to also call `unsubscribe()` on `hbSubscription` in `ConnectionImpl.close()`

If you want I'll contribute a patch here. Doesn't look like I can currently do that

**Comment by jeffkirby** [comment]
I created a pull request
https://github.com/nats-io/java-nats-streaming/pull/27

**Comment by mcqueary** [comment]
Hi Jeff -- thanks, I'll have a look.

**Comment by mcqueary** [comment]
Closed by #27
